File: javascript/lib/sumUpTo.js
// const sumUpTo = number => {
//   let result = 0;

//   let i = 1;
//   while (i <= number) {
//     result += i;
//     i++;
//   }

//   return result;
// };

const sumUpTo = number => {
  if (number === 0) {
    return 0;
  } else {
    return number + sumUpTo(number - 1);
  }
};

export default sumUpTo;

File: ruby/lib/sum_up_to.rb
# def sum_up_to(number)
#   result = 0

#   i = 0
#   while i <= number
#     result += i
#     i += 1
#   end

#   result
# end

def sum_up_to(number)
  if number.zero?
    0
  else
    number + sum_up_to(number.pred)
  end
end
